### PR TITLE
PMD 7.7.0 and fix existing pmd issues

### PR DESCRIPTION
**Pull request by LarsV123**

### CODE REVIEW

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/DynamoDbRetryWrapper.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import software.amazon.awssdk.services.dynamodb.model.PutRequest;
 import software.amazon.awssdk.services.dynamodb.model.WriteRequest;
 
+@SuppressWarnings("PMD.DoNotUseThreads")
```
Comment (brinxmat) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/DynamoDbRetryWrapper.java:17]: Is this a spurious hit? IIRC this should be reported to PMD.

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/dto/CandidateDto.java [hunk 2]
```diff
@@ -16,16 +16,21 @@
...
+                           PeriodStatusDto period,
+                           String status) implements JsonSerializable {
+
+    public static final String NVI_CANDIDATE = "NviCandidate";
+
+    public static Builder builder() {
+        return new Builder();
+    }
 
     //TODO: To remove when frontend has been updated to use "period" property
```
Comment (brinxmat) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/dto/CandidateDto.java:35]: Is this TO DONE? @monaullah

File: nvi-commons/src/test/java/no/sikt/nva/nvi/common/MigrationTests.java [hunk 3]
```diff
@@ -32,9 +33,9 @@ public class MigrationTests extends LocalDynamoTest {
 
     @BeforeEach
     public void setUp() {
-        var localDynamo = initializeTestDatabase();
-        candidateRepository = new CandidateRepository(localDynamo);
-        periodRepository = new PeriodRepository(localDynamo);
+        DynamoDbClient dynamodb = initializeTestDatabase();
```
Comment (brinxmat) [inline nvi-commons/src/test/java/no/sikt/nva/nvi/common/MigrationTests.java:36]: We can't use `var` here`

File: nvi-commons/src/test/java/no/sikt/nva/nvi/common/service/CandidateApprovalDtoTest.java [hunk 4]
```diff
@@ -97,7 +98,7 @@ void shouldUpdateStatusWhenUpdateStatusRequestValid(ApprovalStatus oldStatus, Ap
     }
 
     @ParameterizedTest
-    @EnumSource(value = ApprovalStatus.class, names = {"REJECTED", "APPROVED"})
+    @EnumSource(value = ApprovalStatus.class, names = {"REJECTED", APPROVED})
```
Comment (brinxmat) [inline nvi-commons/src/test/java/no/sikt/nva/nvi/common/service/CandidateApprovalDtoTest.java:101]: A bit weird? only some strings need to be constants in this file?

File: nvi-commons/src/test/java/no/sikt/nva/nvi/common/utils/BatchScanUtilTest.java [hunk 5]
```diff
@@ -151,7 +151,7 @@ private List<Map<String, AttributeValue>> getCandidatesInOrder() {
         return localDynamo.scan(ScanRequest.builder().tableName(ApplicationConstants.NVI_TABLE_NAME).build())
                    .items()
                    .stream()
-                   .filter(a -> a.get("type").s().equals("CANDIDATE"))
+                   .filter(a -> "CANDIDATE".equals(a.get("type").s()))
```
Comment (brinxmat) [inline nvi-commons/src/test/java/no/sikt/nva/nvi/common/utils/BatchScanUtilTest.java:154]: Odd that it isn't as insistent on constants here

File: nvi-commons/src/test/java/no/sikt/nva/nvi/common/utils/JsonUtilsTest.java [hunk 6]
```diff
@@ -4,65 +4,59 @@
...
+        var randomJsonNode = objectMapper.createObjectNode().put(FIELD_NAME, fieldValue);
+        assertEquals(extractOptJsonNodeTextValue(randomJsonNode, "/" + FIELD_NAME).get(), fieldValue);
     }
 
     @Test
     void shouldReturnNullIfJsonPointerPointsToMissingNode() {
-        var fieldName = "fieldName";
         var randomJsonNode = objectMapper.createObjectNode();
-        assertThat(JsonUtils.extractJsonNodeTextValue(randomJsonNode, "/" + fieldName), nullValue());
+        assertNull(JsonUtils.extractJsonNodeTextValue(randomJsonNode, "/" + FIELD_NAME));
```
Comment (brinxmat) [inline nvi-commons/src/test/java/no/sikt/nva/nvi/common/utils/JsonUtilsTest.java:34]: the "/" here is a thing, JSON_PATH_DELIMITER? makes it clear what it is?

**Review by brinxmat — CHANGES_REQUESTED** [review]
A few small inconsistencies seen from outside what PMD thinks is OK

File: config/pmd/ruleset.xml [hunk 7]
```diff
@@ -39,33 +47,19 @@
...
-    <!-- This rule does not allow us the following:
-            private static final Logger logger=...
-         It requires:
-            private static final Logger LOGGER=... -->
     <exclude name="FieldNamingConventions"/>
 
     <!-- Sometimes it is useful in debugging to assign the return value to a local variable  -->
     <exclude name="UnnecessaryLocalBeforeReturn"/>
-
+    <exclude name="JUnitTestsShouldIncludeAssert"/>
```
[the remark this answers is not in the document]
Comment (monaullah) [reply config/pmd/ruleset.xml:62]: Uups, intention was to exclude _JUnit assertions should include a message_ (The one failing now). Will include

File: nvi-commons/src/test/java/no/sikt/nva/nvi/common/service/CandidateApprovalDtoTest.java (hunk 4 above)
Comment (monaullah) [reply nvi-commons/src/test/java/no/sikt/nva/nvi/common/service/CandidateApprovalDtoTest.java:101]: The ones that are repeated over four times

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/client/model/Organization.java [hunk 8]
```diff
@@ -11,21 +11,14 @@
...
 
-public record Organization(@JsonProperty(ID) URI id,
-                           @JsonProperty(PART_OF) List<Organization> partOf,
-                           @JsonProperty(HAS_PART) List<Organization> hasPart,
-                           @JsonProperty(LABELS) Map<String, String> labels,
-                           @JsonProperty(TYPE) String type,
-                           @JsonProperty(CONTEXT) Object context)
+public record Organization(@JsonProperty("id") URI id,
+                           @JsonProperty("partOf") List<Organization> partOf,
+                           @JsonProperty("hasPart") List<Organization> subUnits,
```
[the remark this answers is not in the document]
Comment (monaullah) [reply nvi-commons/src/main/java/no/sikt/nva/nvi/common/client/model/Organization.java]: https://github.com/BIBSYSDEV/nva-nvi/pull/403/commits/d7d1d4ee235c08e5a337811a2931c05d84fe261e

File: config/pmd/ruleset.xml [hunk 9]
```diff
@@ -24,10 +26,16 @@
     <exclude name="SystemPrintln"/>
     <exclude name="GuardLogStatement"/>
     <exclude name="AvoidPrintStackTrace"/>
+    <exclude name="UnitTestAssertionsShouldIncludeMessage"/>
+    <exclude name="UnitTestContainsTooManyAsserts"/>
+    <exclude name="JUnit5TestShouldBePackagePrivate"/>
```
[the remark this answers is not in the document]
Comment (monaullah) [reply config/pmd/ruleset.xml]: https://github.com/BIBSYSDEV/nva-nvi/pull/403/commits/33f5f2740f674b050d9cf97cedb0224d1c79cccf

File: config/pmd/ruleset.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (monaullah) [reply config/pmd/ruleset.xml]: https://github.com/BIBSYSDEV/nva-nvi/pull/403/commits/489de8ba34eebf8c884da30a6ba0f6e3a2b6dbee

File: config/pmd/ruleset.xml [hunk 10]
```diff
@@ -24,10 +26,16 @@
...
 
     <!-- In TableManager we map ResourceNotFoundException to TableNotFoundException -->
     <exclude name="PreserveStackTrace"/>
 
+    <!-- //TODO: This is complaining on private methods that are used?  -->
+    <exclude name="UnusedPrivateMethod"/>
```
[the remark this answers is not in the document]
Comment (monaullah) [reply config/pmd/ruleset.xml:36]: [NVI PMD Issue: UnusedPrivateMethod](https://sikt.atlassian.net/browse/NP-48102)

**Review by monaullah — APPROVED** [review]

File: config/pmd/ruleset.xml (hunk 10 above)
[the remark this answers is not in the document]
Comment (monaullah) [reply config/pmd/ruleset.xml:36]: https://github.com/pmd/pmd/issues/5083

**Pull request merged**